CUDA backend produces inconsistent results for jax.numpy.linalg.inv

With a P4 GPU, the unit test fails out of the box for me:

```
======================================================================
ERROR: jax.test_util.format_test_name_suffix
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/adam/dev/jax/venv2/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
TypeError: format_test_name_suffix() takes exactly 3 arguments (0 given)
  
======================================================================
FAIL: testInv_shape=float32[200,200] (linalg_test.NumpyLinalgTest)
testInv_shape=float32[200,200](dtype=dtype('float32'), shape=(200, 200), rng=<functools.partial>)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/adam/dev/jax/venv2/local/lib/python2.7/site-packages/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/home/adam/dev/jax/venv2/local/lib/python2.7/site-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/home/adam/dev/jax/venv2/local/lib/python2.7/site-packages/absl/testing/parameterized.py", line 256, in bound_param_test
    test_method(self, **testcase_params)
  File "/home/adam/dev/jax/tests/linalg_test.py", line 133, in testInv
    self._CompileAndCheck(np.linalg.inv, args_maker, check_dtypes=True)
  File "/home/adam/dev/jax/jax/test_util.py", line 429, in _CompileAndCheck
    self.assertAllClose(python_ans, monitored_ans, check_dtypes, rtol, atol)
  File "/home/adam/dev/jax/jax/test_util.py", line 408, in assertAllClose
    self.assertArraysAllClose(x, y, check_dtypes, atol=atol, rtol=rtol)
  File "/home/adam/dev/jax/jax/test_util.py", line 386, in assertArraysAllClose
    raise self.failureException(msg)
AssertionError: Arguments x and y not equal to tolerance atol=0.01, rtol=0.01:
x:
[[ 0.00618084  0.01218688  0.0251631  ...  0.00313068 -0.04788522
   0.0435191 ]
 [-0.0086758   0.01550603  0.01697903 ... -0.02351537 -0.02903247
  -0.00979138]
 [-0.06120062 -0.01663456  0.00417364 ... -0.00088828  0.02902133
   0.02269716]
 ...
 [-0.01421827 -0.01083164 -0.00547409 ...  0.01480494  0.00061655
   0.00666774]
 [ 0.0018405  -0.01860033  0.00488692 ...  0.00800579 -0.00021123
   0.00595227]
 [ 0.02060174 -0.01633311 -0.02736879 ... -0.0021769  -0.00297758
  -0.02157698]]
y:
[[ 0.03693918 -0.02123345  0.00972237 ... -0.01746454 -0.07871114
   0.01237083]
 [-0.0037162   0.01538941  0.02761243 ... -0.0108461  -0.026155
   0.00196449]
 [-0.08331504 -0.03236853 -0.05631161 ...  0.01518264  0.07201606
   0.0681076 ]
 ...
 [-0.01517212 -0.02950297 -0.04371436 ...  0.02140163  0.01937166
   0.00679701]
 [ 0.00105234 -0.04179138 -0.00391355 ... -0.00963392 -0.02027218
  -0.01098038]
 [ 0.02950978 -0.01039723 -0.02611141 ...  0.00717233  0.02393566
  -0.02157698]]
```

I dug into it a little bit and the failure was taking place in the _CompileAndCheck() test routine.  Specifically, jitted and non-jitted versions of the function give different answers:

```
def myinv(a):
    q, r = jnp.linalg.lax_linalg.qr(a)
    return jnp.linalg.lax_linalg.triangular_solve(r, q.T, lower=False, left_side=True)

jitted_myinv = jit(myinv)
print(np.max(np.abs(myinv(bad), jitted_myinv(bad)))
```

where bad is the random array dumped from the unit test, but generating random (200,200) matrices can cause it.

Interestingly, if they're pulled apart:

```
def myqr(a):
    q, r = jnp.linalg.lax_linalg.qr(a)
    return q, r

def myts(q, r):
    return jnp.linalg.lax_linalg.triangular_solve(r, q.T, lower=False, left_side=True)

jitted_myqr = jit(myqr)
jitted_myts = jit(myts)

q, r = myqr(bad)
res1 = myts(q, r)

qc, rc = jitted_myqr(bad)
res2 = myts(qc,qr)

print(np.allclose(res1,res2))
```
The problem does not surface.

Since the problem doesn't occur with the CPU (Host) backend, it seems like maybe an XLA bug? I did dump the unoptimized HLO from both the Host and CUDA backends and they were identical, but they diverge when dumping after the optimizer runs (I guess the platform specific optimizations start to take place there?).

Also, what's the best way to debug these issues?  I see that functionality to dump jaxprs has been added?  Any other tricks for dumping useful information and visualizing it?  (Looking at the  output from protoc for the HLO was not the most fun for trying to make sense of it)

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -17,10 +17,10 @@ http_archive(
 #    and update the sha256 with the result.
 http_archive(
    name = "org_tensorflow",
-   sha256 = "f10fbc64cc4cd5ada4ebddcdd26f8f76cd97e6aff8e25956172ee18f98a705ea",
-   strip_prefix = "tensorflow-787c2a684e39d9c21525c07da3e56bc68621e9b7",
+   sha256 = "2434ae2aebbf681534fca88f3ec547f49bab3968fb838b6589c3f1ccbde617ee",
+   strip_prefix = "tensorflow-38c91321421694432117b077294df43aa31d1193",
    urls = [
-       "https://github.com/tensorflow/tensorflow/archive/787c2a684e39d9c21525c07da3e56bc68621e9b7.tar.gz",
+       "https://github.com/tensorflow/tensorflow/archive/38c91321421694432117b077294df43aa31d1193.tar.gz",
    ],
 )
 

